pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,390 @@
+    mark_median_entries.extend(non_usd_mark_median_entries?);
+    Ok(mark_median_entries)
+}
+
+async fn compute_mark_median_entries_for_usd_pairs(
+    state: &AppState,
+    usd_pairs: Vec<String>,
+) -> Result<Vec<MedianEntryWithComponents>, EntryError> {
+    if usd_pairs.is_empty() {
+        return Ok(vec![]);
+    }
+    get_current_median_entries_with_components(
+        &state.timescale_pool,
+        &usd_pairs,
+        DataType::PerpEntry,
+    )
+    .await
+    .map_err(|e| e.to_entry_error(&usd_pairs.join(",")))
+}
+
+async fn compute_mark_median_entries_for_non_usd_pairs(
+    state: &AppState,
+    non_usd_pairs: Vec<String>,
+) -> Result<Vec<MedianEntryWithComponents>, EntryError> {
+    if non_usd_pairs.is_empty() {
+        return Ok(vec![]);
+    }
+    let usd_pairs: Vec<String> = non_usd_pairs
+        .iter()
+        // safe to unwrap since we are sure that the pairs are in the format of "XXX/YYY"
+        .map(|pair| format!("{}/USD", pair.split('/').last().unwrap()))
+        .collect();
+
+    let (usd_pairs_spot_median_entries, pairs_perp_median_entries) = tokio::join!(
+        get_current_median_entries_with_components(
+            &state.timescale_pool,
+            &usd_pairs,
+            DataType::SpotEntry,
+        ),
+        get_current_median_entries_with_components(
+            &state.timescale_pool,
+            &non_usd_pairs,
+            DataType::PerpEntry,
+        )
+    );
+
+    let mut mark_median_entries = vec![];
+    let usd_pairs_spot_median_entries = usd_pairs_spot_median_entries?;
+    let pairs_perp_median_entries = pairs_perp_median_entries?;
+
+    for perp_median_entry in pairs_perp_median_entries {
+        let related_usd_spot = format!(
+            "{}/USD",
+            // safe to unwrap since we are sure that the pairs are in the format of "XXX/YYY"
+            perp_median_entry.pair_id.split('/').last().unwrap()
+        );
+
+        let spot_usd_median_entry = usd_pairs_spot_median_entries
+            .iter()
+            .find(|spot_median_entry| spot_median_entry.pair_id == related_usd_spot)
+            .ok_or(EntryError::InternalServerError)?;
+
+        let perp_pair_price = perp_median_entry.median_price.clone();
+        let spot_usd_price = spot_usd_median_entry.median_price.clone();
+        let mark_price = perp_pair_price / spot_usd_price;
+
+        let mut components = perp_median_entry.components;
+        components.extend(spot_usd_median_entry.components.clone());
+
+        let mark_median_entry = MedianEntryWithComponents {
+            // TODO: do we need to change eg. BTC/USDT to BTC/USD here?
+            pair_id: perp_median_entry.pair_id.clone(),
+            median_price: mark_price,
+            components,
+        };
+        mark_median_entries.push(mark_median_entry);
+    }
+
+    Ok(mark_median_entries)
+}
+
+/// Sign the median price with the passed signer and return the signature 0x prefixed.
+fn sign_median_price(
```

Comment:
move to `utils/signing.rs` also we could leverage a `Signable` trait that we implement on the data types we want to be signable

Reply:
Agreed - created an issue for this: #61

Reply:
Yeah no, bad idea to stack too much TODOs.

I implemented it directly now